pkg/operation/botanist/component/nodeproblemdetector/node_problem_detector_test.go

```diff
@@ -379,32 +379,62 @@ status: {}
 			Expect(string(managedResourceSecret.Data["serviceaccount__kube-system__node-problem-detector.yaml"])).To(Equal(serviceAccountYAML))
 			Expect(string(managedResourceSecret.Data["clusterrole____node-problem-detector.yaml"])).To(Equal(clusterRoleYAML))
 			Expect(string(managedResourceSecret.Data["clusterrolebinding____node-problem-detector.yaml"])).To(Equal(clusterRoleBindingYAML))
-			Expect(string(managedResourceSecret.Data["clusterrole____node-problem-detector-psp.yaml"])).To(Equal(clusterRolePSPYAML))
-			Expect(string(managedResourceSecret.Data["clusterrolebinding____node-problem-detector-psp.yaml"])).To(Equal(clusterRoleBindingPSPYAML))
-			Expect(string(managedResourceSecret.Data["podsecuritypolicy____node-problem-detector.yaml"])).To(Equal(podSecurityPolicyYAML))
 		})
 
 		Context("w/o apiserver host, w/o vpaEnables", func() {
-			It("should successfully deploy all resources", func() {
-				Expect(string(managedResourceSecret.Data["daemonset__kube-system__node-problem-detector.yaml"])).To(Equal(daemonsetYAMLFor("", false)))
+			Context("PSP is disabled", func() {
+				BeforeEach(func() {
+					values.PSPDisabled = true
+				})
+				It("should successfully deploy all resources", func() {
+					Expect(string(managedResourceSecret.Data["daemonset__kube-system__node-problem-detector.yaml"])).To(Equal(daemonsetYAMLFor("", false)))
+				})
+			})
+
+			Context("PSP is not disabled", func() {
+				BeforeEach(func() {
+					values.PSPDisabled = false
+				})
+				It("should successfully deploy all resources", func() {
+					Expect(string(managedResourceSecret.Data["daemonset__kube-system__node-problem-detector.yaml"])).To(Equal(daemonsetYAMLFor("", false)))
+					Expect(string(managedResourceSecret.Data["clusterrolebinding____gardener.cloud_psp_node-problem-detector.yaml"])).To(Equal(clusterRoleBindingPSPYAML))
+					Expect(string(managedResourceSecret.Data["clusterrole____gardener.cloud_psp_kube-system_node-problem-detector.yaml"])).To(Equal(clusterRolePSPYAML))
+					Expect(string(managedResourceSecret.Data["podsecuritypolicy____node-problem-detector.yaml"])).To(Equal(podSecurityPolicyYAML))
+				})
 			})
 		})
 
-		Context("w/ apiserver host,w/ vpaEnables", func() {
+		Context("w/ apiserver host,w/ vpaEnabled", func() {
 			var (
 				apiserverHost = "apiserver.host"
 				vpaEnabled    = true
 			)
-
 			BeforeEach(func() {
 				values.APIServerHost = &apiserverHost
 				values.VPAEnabled = vpaEnabled
-				component = New(c, namespace, values)
 			})
 
-			It("should successfully deploy all resources", func() {
-				Expect(string(managedResourceSecret.Data["verticalpodautoscaler__kube-system__node-problem-detector.yaml"])).To(Equal(vpaYAML))
-				Expect(string(managedResourceSecret.Data["daemonset__kube-system__node-problem-detector.yaml"])).To(Equal(daemonsetYAMLFor(apiserverHost, vpaEnabled)))
+			Context("PSP is disabled", func() {
```

Comment:
We add PSP tests for the cases `"w/o apiserver host, w/o vpaEnables"` and `"w/ apiserver host,w/ vpaEnabled"`. The PSP tests don't have much to do with the apiserver host and the VPA, right? I was wondering instead of adding as a subcontext the PSP tests (PSP disabled/enabled) is it possible rather to move the PSP testing to a separate context? For the PSP tests it should not matter much if the VPA is enabled or apiserver host is used.